includes/abstracts/abstract-wc-payment-gateway-payex.php

```diff
@@ -124,6 +124,57 @@ protected function log( $message, $level = 'notice' ) {
 	 * @throws \Exception
 	 */
 	public function request( $method, $url, $params = array() ) {
+		$client = new \PayEx\Api\Client();
+		$client->setMerchantToken( $this->merchant_token );
+		$client->setMode( $this->testmode === 'yes' ? \PayEx\Api\Client::MODE_TEST : \PayEx\Api\Client::MODE_PRODUCTION );
+
+		$start = microtime( true );
+		if ( $this->debug === 'yes' ) {
+			$this->log( sprintf( 'Request: %s %s %s', $method, $url, json_encode( $params, JSON_PRETTY_PRINT ) ) );
+		}
+
+		try {
+			/** @var \PayEx\Api\Response $response */
+			$response = $client->request( $method, $url, $params );
+			$result   = $response->toArray();
+			$this->response_body = $client->getLastResponse();
+
+			if ( $this->debug === 'yes' ) {
+				$time = microtime( true ) - $start;
+				$this->log( sprintf( '[%.4F] Response: %s', $time, $response->getBody() ) );
+			}
+
+			return $result;
+		} catch ( \PayEx\Api\Exception $e ) {
+			if ( $this->debug === 'yes' ) {
+				$time = microtime( true ) - $start;
+				$this->log( sprintf( '[%.4F] Exception: %s', $time, $e->getMessage() ) );
+			}
+
+			$this->response_body = $client->getLastResponse();
+
+			// Message for invalid Msisdn
+			if ( ( strpos( $this->response_body, 'The Msisdn is not valid' ) !== false ) ||
```

Comment:
Please don't search through `response_body` after human readable error messages. Use the `type` or `problems` properties of the `application/problem+json` response.